### PR TITLE
chore(MeshHTTPRoute): default catch all path in route match

### PR DESCRIPTION
**Pull request by bartsmykla**
According to Envoy docs, match must have precisely one of prefix, path, safe_regex, connect_matcher, path_separated_prefix, path_match_policy set, so when policy doesn't specify explicit type of matching, we are assuming "catch all" path (any path starting with "/").

Closes: https://github.com/kumahq/kuma/issues/6758

### Checklist prior to review

- [x] [Link to relevant issue][1] as well as docs and UI issues
  - https://github.com/kumahq/kuma/issues/6758
- [x] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS
  - It won't
- [x] Tests (Unit test, E2E tests, manual test on universal and k8s)
  - Updated golden files for unit tests
- [x] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)?
  - There is no need
- [x] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? 
  - There is no need
- [x] Do you need to explicitly set a [`> Changelog:` entry here](../blob/master/CONTRIBUTING.md#submitting-a-patch) or add a `ci/` label to run fewer/more tests?
> Changelog: chore(MeshHTTPRoute): assume default catch all path (any path starting with "/") in route match when not explicitly set

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: pkg/plugins/policies/meshhttproute/api/v1alpha1/helpers.go [hunk 1]
```diff
@@ -5,7 +5,33 @@ type PolicyDefault struct {
...
 
 func (x *To) GetDefault() interface{} {
+	var rules []Rule
+
+	for _, rule := range x.Rules {
+		var matches []Match
+
+		for _, match := range rule.Matches {
+			if match.Path == nil {
+				// According to Envoy docs, match must have precisely one of
```
Comment (michaelbeaumont) [inline pkg/plugins/policies/meshhttproute/api/v1alpha1/helpers.go:15]: can we handle this in the XDS generation instead? It makes me somewhat nervous returning resources that the user didn't configure and wouldn't see match if they run `kubectl get`

File: pkg/plugins/policies/meshhttproute/api/v1alpha1/helpers.go [hunk 2]
```diff
@@ -5,7 +5,33 @@ type PolicyDefault struct {
...
+				// prefix, path, safe_regex, connect_matcher,
+				// path_separated_prefix, path_match_policy set, so when policy
+				// doesn't specify explicit type of matching, we are assuming
+				// "catch all" path (any path starting with "/").
+				match.Path = &PathMatch{
+					Value: "/",
+					Type:  PathPrefix,
+				}
+			}
```
Comment (jakubdyszkiewicz) [inline pkg/plugins/policies/meshhttproute/api/v1alpha1/helpers.go:24]: shouldn't this logic be placed where we generate Envoy config? It's sort of Envoy implem details that it always requires path matcher

File: pkg/plugins/policies/meshhttproute/api/v1alpha1/helpers.go (hunk 1 above)
Comment (bartsmykla) [reply pkg/plugins/policies/meshhttproute/api/v1alpha1/helpers.go:15]: you are right

File: pkg/plugins/policies/meshhttproute/api/v1alpha1/helpers.go (hunk 2 above)
Comment (bartsmykla) [reply pkg/plugins/policies/meshhttproute/api/v1alpha1/helpers.go:24]: you are right

File: pkg/plugins/policies/meshhttproute/api/v1alpha1/helpers.go (hunk 1 above)
Comment (bartsmykla) [reply pkg/plugins/policies/meshhttproute/api/v1alpha1/helpers.go]: addressed

File: pkg/plugins/policies/meshhttproute/api/v1alpha1/helpers.go (hunk 2 above)
Comment (bartsmykla) [reply pkg/plugins/policies/meshhttproute/api/v1alpha1/helpers.go]: addressed

File: pkg/plugins/policies/meshhttproute/plugin/v1alpha1/listeners.go [hunk 3]
```diff
@@ -146,8 +146,26 @@ func prepareRoutes(
...
+
+		for _, match := range rule.Matches {
+			if match.Path == nil {
+				// According to Envoy docs, match must have precisely one of
+				// prefix, path, safe_regex, connect_matcher,
+				// path_separated_prefix, path_match_policy set, so when policy
+				// doesn't specify explicit type of matching, we are assuming
+				// "catch all" path (any path starting with "/").
+				match.Path = &api.PathMatch{
+					Value: "/",
```
Comment (michaelbeaumont) [inline pkg/plugins/policies/meshhttproute/plugin/v1alpha1/listeners.go:159]: nit: could be defined privately in this file and used here and above at `catchAllMatch`

File: pkg/plugins/policies/meshtimeout/plugin/testdata/no-route-level-timeouts.gateway.route.golden.yaml [hunk 4]
```diff
@@ -15,13 +15,13 @@ virtualHosts:
       idleTimeout: 5s
       weightedClusters:
         clusters:
-        - name: backend-26cb64fa4e85e7b7
+        - name: other-service-7ab0707c3913adf1
```
Comment (michaelbeaumont) [inline pkg/plugins/policies/meshtimeout/plugin/testdata/no-route-level-timeouts.gateway.route.golden.yaml:18]: rebase on `master`

File: pkg/plugins/policies/meshhttproute/plugin/v1alpha1/listeners.go (hunk 3 above)
Comment (bartsmykla) [reply pkg/plugins/policies/meshhttproute/plugin/v1alpha1/listeners.go]: addressed

File: pkg/plugins/policies/meshtimeout/plugin/testdata/no-route-level-timeouts.gateway.route.golden.yaml (hunk 4 above)
Comment (bartsmykla) [reply pkg/plugins/policies/meshtimeout/plugin/testdata/no-route-level-timeouts.gateway.route.golden.yaml]: done

File: pkg/plugins/policies/meshhttproute/plugin/v1alpha1/listeners.go [hunk 5]
```diff
@@ -146,8 +142,23 @@ func prepareRoutes(
...
+		var matches []api.Match
+
+		for _, match := range rule.Matches {
+			if match.Path == nil {
+				// According to Envoy docs, match must have precisely one of
+				// prefix, path, safe_regex, connect_matcher,
+				// path_separated_prefix, path_match_policy set, so when policy
+				// doesn't specify explicit type of matching, we are assuming
+				// "catch all" path (any path starting with "/").
+				match.Path = catchAllPathMatch
```
Comment (michaelbeaumont) [inline pkg/plugins/policies/meshhttproute/plugin/v1alpha1/listeners.go:154]: hmm the sharing via might be dangerous, maybe define as value and use `pointer.To` in both places?

Comment (bartsmykla) [reply pkg/plugins/policies/meshhttproute/plugin/v1alpha1/listeners.go]: done

**Review by michaelbeaumont — APPROVED** [review]

**Pull request merged**